filechooser-portal/Main.vala

```diff
@@ -343,10 +333,58 @@ public class Files.FileChooserPortal : Object {
         yield;
 
         dialogs.remove (parent_window);
+        dialog.destroy ();
+
         response = _response;
         results = _results;
     }
 
+    private async bool check_overwrite_uri (string? uri) {
+        if (uri == null) {
+            return false;
+        }
+
+        var file = GLib.File.new_for_uri (uri);
+        if (file.query_exists ()) {
+            var display_name = file.get_basename ();
+            var primary = _("Replace “%s”?").printf (display_name);
+            unowned var secondary = _("Replacing this file will overwrite its current contents");
+            if (file.query_file_type (FileQueryInfoFlags.NOFOLLOW_SYMLINKS) == FileType.SYMBOLIC_LINK) {
+                try {
+                    var info = file.query_info (FileAttribute.STANDARD_SYMLINK_TARGET, FileQueryInfoFlags.NONE);
+                    primary = _("Replace “%s”?").printf (info.get_symlink_target ());
+                } catch (Error e) {
+                    warning ("Could not get info for %s", file.get_uri ());
+                    primary = _("Replace the target of “%s”?").printf (display_name);
+                }
+
+                secondary = _("Replacing the target file for this link will overwrite its current contents.");
+            }
+
+            var replace_dialog = new Granite.MessageDialog.with_image_from_icon_name (
+                primary, secondary, "dialog-warning", Gtk.ButtonsType.CANCEL
+            );
+
+            replace_dialog.modal = true;
```

Comment:
```suggestion
            ) {
                modal = true,
                transient_for = parent
            };
```

The dialog should be modal, and I think that it should be modal to the `FileChooserDialog`. Currently the replace dialog appears in the window switcher, which it shouldn't. This fixes that.

This change would also need some changes elsewhere to have a `FileChooserDialog` reference to work with. See other comments in this review.